### PR TITLE
[MI-3828] Add feature to show linked channels and search the linked channels

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary

- Added feature to show linked channels
- Added search field in the RHS to search for linked channels
- Added tooltip on the text displaying RHS data

[screen-capture (36).webm](https://github.com/Brightscout/mattermost-plugin-msteams-sync/assets/100013900/46a4af00-ddea-4456-a875-698bac22d052)

### CODE REVIEW

File: webapp/src/hooks/usePreviousState.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import {useEffect, useRef} from 'react';
+
+function usePreviousState(value: Record<string, string>) {
```
Comment (SaurabhSharma-884) [inline webapp/src/hooks/usePreviousState.ts:3]: Add doc

File: webapp/src/styles/_utils.scss [hunk 2]
```diff
@@ -104,6 +104,14 @@
     margin-top: 0;
 }
 
+.pt-0 {
```
Comment (SaurabhSharma-884) [inline webapp/src/styles/_utils.scss:107]: Can add these in @each loop

**Review by SaurabhSharma-884 — APPROVED** [review]

File: webapp/src/containers/Rhs/Rhs.container.tsx [hunk 3]
```diff
@@ -1,59 +1,319 @@
...
+    useEffect(() => {
+        if (firstRender) {
+            return;
+        }
+
+        const timer = setTimeout(() => {
+            resetStates();
+        }, debounceFunctionTimeLimit);
+
+        /* eslint-disable consistent-return */
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply webapp/src/containers/Rhs/Rhs.container.tsx:93]: discussed

File: webapp/src/containers/Rhs/Rhs.container.tsx [hunk 4]
```diff
@@ -1,59 +1,321 @@
...
+    const hasMoreLinkedChannels = useMemo<boolean>(() => (
+        (totalLinkedChannels.length - (paginationQueryParams.page * defaultPerPage) === defaultPerPage)
+    ), [totalLinkedChannels]);
+
+    // Show disconnect dialog component
+    const {DialogComponent, showDialog, hideDialog} = useDialog({
+        onSubmitHandler: disconnectUser,
+        onCloseHandler: () => {
+            hideDialog();
+        },
```
Comment (avas27JTG) [inline webapp/src/containers/Rhs/Rhs.container.tsx:74]:
```suggestion
        onCloseHandler: () => hideDialog(),
```

File: webapp/src/containers/Rhs/Rhs.container.tsx [hunk 5]
```diff
@@ -1,59 +1,321 @@
...
+
+    // Update total linked channels after completion of the api to get linked channels
+    useApiRequestCompletionState({
+        serviceName: pluginApiServiceConfigs.getLinkedChannels.apiServiceName,
+        payload: getLinkedChannelsParams as SearchLinkedChannelParams,
+        handleSuccess: () => {
+            if (linkedChannels) {
+                setTotalLinkedChannels([...totalLinkedChannels, ...(linkedChannels as ChannelLinkData[])]);
+            }
+            if (firstRender && !paginationQueryParams.page) {
```
Comment (avas27JTG) [inline webapp/src/containers/Rhs/Rhs.container.tsx:126]: for checking if RHS is opened for the first time or not we can make use of `view.rhs.isOpen` and `usePreviousState`

see if it works

File: webapp/src/containers/Rhs/Rhs.container.tsx [hunk 6]
```diff
@@ -1,59 +1,321 @@
...
+                setFirstRender(false);
+            }
+        },
+    });
+
+    // Disconnect user and show alerts on completion of the api to disconnect user
```
Comment (avas27JTG) [inline webapp/src/containers/Rhs/Rhs.container.tsx:132]:
```suggestion
    // Disconnect a user and show alerts on completion of the api to disconnect the user
```

File: webapp/src/components/LinkedChannelCard/LinkedChannelCard.styles.scss [hunk 7]
```diff
@@ -26,5 +26,17 @@
...
+    &__body {
+      width: 95%;
+    }
+
+    &__entity-label {
+      white-space: nowrap;
+      overflow: hidden;
+      text-overflow: ellipsis;
+      max-width: 45%;
+      line-height: inherit;
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply webapp/src/components/LinkedChannelCard/LinkedChannelCard.styles.scss:39]: updated the styles

File: webapp/src/containers/Rhs/Rhs.container.tsx (hunk 5 above)
Comment (ayusht2810) [reply webapp/src/containers/Rhs/Rhs.container.tsx:121]: I think `views.rhs.open` will also get true when other plugins rhs also get opened. I don't think we can use it.

**Review by avas27JTG — APPROVED** [review]

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**